### PR TITLE
add a basic fallback for test/REQUIRE

### LINKED ISSUE #1596: Add a super simple REQUIRE parser
A lot of packages fail their tests on 1.4 because we removed the support for REQUIRE files. The problem is when packages have a `test/REQUIRE` then those packages are not included when running the tests.

I suggest we add back a super simple REQUIRE parser that just reads the packages, ignores all version bounds and adds those packages. This should fix most cases since people don't really use version bounds in REQUIRE, especially test/REQUIRE.

### PR DESCRIPTION
**Pull request by KristofferC**
Fixes #1596

### CODE REVIEW

File: src/Operations.jl [hunk 1]
```diff
@@ -1301,7 +1301,11 @@ function sandbox_preserve(ctx::Context, target::PackageSpec, test_project::Strin
     end
     # preserve important nodes
     keep = [target.uuid]
-    append!(keep, collect(values(read_project(test_project).deps)))
+    test_project_deps = read_project(test_project).deps
+    for (pkg, uuid) in test_project_deps
```
Comment (KristofferC) [inline src/Operations.jl]: I guess this is what CI doesn't like. I did this so I could get the new test packages into the manifest but perhaps this adds too much.

Comment (00vareladavid) [reply src/Operations.jl]: We can get rid of this block of changes. Any test packages that don't exist in the active manifest will be resolved later on.

File: src/Operations.jl [hunk 2]
```diff
@@ -1384,12 +1388,43 @@ function update_package_test!(pkg::PackageSpec, entry::PackageEntry)
...
+    end
+    return packages
+end
+
 # "targets" based test deps -> "test/Project.toml" based deps
 function gen_target_project(ctx::Context, pkg::PackageSpec, source_path::String, target::String)
     test_project = Types.Project()
     if projectfile_path(source_path; strict=true) === nothing
         # no project file, assuming this is an old REQUIRE package
         test_project.deps = ctx.env.manifest[pkg.uuid].deps
```
Comment (00vareladavid) [inline src/Operations.jl]: We might need a `copy` here

File: src/Operations.jl (hunk 1 above)
Comment (KristofferC) [reply src/Operations.jl]: I originally didn't have it there but then I got:

```
(Pkg) pkg> test EnglishText
┌ Warning: using test/REQUIRE files is deprecated and current support is lacking in some areas
└ @ Pkg.Operations ~/Pkg.jl/src/Operations.jl:1414
    Testing EnglishText
ERROR: KeyError: key "Compat" not found
Stacktrace:
 [1] getindex at ./dict.jl:477 [inlined]
 [2] #54 at /Users/kristoffer/Pkg.jl/src/manifest.jl:209 [inlined]
 [3] _all(::Pkg.Types.var"#54#57"{Dict{String,Bool}}, ::Dict{String,Base.UUID}, ::Colon) at ./reduce.jl:729
 [4] all at ./reduce.jl:724 [inlined]
 [5] destructure(::Dict{Base.UUID,Pkg.Types.PackageEntry}) at /Users/kristoffer/Pkg.jl/src/manifest.jl:209
 [6] write_manifest at /Users/kristoffer/Pkg.jl/src/manifest.jl:227 [inlined]
 [7] (::Pkg.Operations.var"#86#90"{Pkg.Operations.var"#97#98"{Bool,Cmd,Cmd,Nothing,Pkg.Types.Context,Array{String,1},String,Pkg.Types.PackageSpec},Pkg.Types.Context,Pkg.Types.PackageSpec,String,Pkg.Types.Project,String})(::String) at /Users/kristoffer/Pkg.jl/src/Operations.jl:1356
 [8] #mktempdir#18(::String, ::typeof(mktempdir), ::Pkg.Operations.var"#86#90"{Pkg.Operations.var"#97#98"{Bool,Cmd,Cmd,Nothing,Pkg.Types.Context,Array{String,1},String,Pkg.Types.PackageSpec},Pkg.Types.Context,Pkg.Types.PackageSpec,String,Pkg.Types.Project,String}, ::String) at ./file.jl:634
 [9] mktempdir(::Function, ::String) at ./file.jl:632 (repeats 2 times)
 [10] sandbox(::Function, ::Pkg.Types.Context, ::Pkg.Types.PackageSpec, ::String, ::String, ::Pkg.Types.Project) at /Users/kristoffer/Pkg.jl/src/Operations.jl:1325
 [11] #test#96(::Bool, ::Cmd, ::Cmd, ::Nothing, ::typeof(Pkg.Operations.test), ::Pkg.Types.Context, ::Array{Pkg.Types.PackageSpec,1}) at /Users/kristoffer/Pkg.jl/src/Operations.jl:1488
 [12] #test at ./none:0 [inlined]
 [13] #test#69(::Bool, ::Nothing, ::Cmd, ::Cmd, ::Base.Iterators.Pairs{Union{},Union{},Tuple{},NamedTuple{(),Tuple{}}}, ::typeof(Pkg.API.test), ::Pkg.Types.Context, ::Array{Pkg.Types.PackageSpec,1}) at /Users/kristoffer/Pkg.jl/src/API.jl:313
 [14] test(::Pkg.Types.Context, ::Array{Pkg.Types.PackageSpec,1}) at /Users/kristoffer/Pkg.jl/src/API.jl:300
 [15] #test#68 at /Users/kristoffer/Pkg.jl/src/API.jl:294 [inlined]
 [16] test(::Array{Pkg.Types.PackageSpec,1}) at /Users/kristoffer/Pkg.jl/src/API.jl:294
```

**Comment by KristofferC** [comment]
@00vareladavid you know this code better than me so feel free to push a fix to this branch if you know one

**Comment by KristofferC** [comment]
As a related note, I think it would be good to have some high level description of what e.g. the sandbox function does in a comment. There are a lot of projects and manifests getting created and `Pkg.resolve` is done in a `try catch` etc. It is kinda hard to figure out the end goal by just reading the code.

**Pull request closed without merging**